File: Backend/Ready2RoadBackend/src/main/java/Ready2road/Persistenza/Model/Venditore.java
package Ready2road.Persistenza.Model;

import com.google.gson.JsonObject;

public class Venditore {
    public String getNomeSocieta() {
        return nomeSocieta;
    }

    public void setNomeSocieta(String nomeSocieta) {
        this.nomeSocieta = nomeSocieta;
    }

    public String getPassword() {
        return password;
    }


    public void setPassword(String password) {
        this.password = password;
    }

    public String getIndirizzoEmail() {
        return indirizzoEmail;
    }

    public void setIndirizzoEmail(String indirizzoEmail) {
        this.indirizzoEmail = indirizzoEmail;
    }

    public Wallet getWallet() {
        return wallet;
    }

    public void setWallet(Wallet wallet) {
        this.wallet = wallet;
    }

    private String nomeSocieta, password, indirizzoEmail;
    private Boolean ban;

    public Boolean isBan() {
        return ban;
    }

    public void setBan(Boolean ban) {
        this.ban = ban;
    }

    private Wallet wallet;

    public JsonObject toJson() {
        JsonObject jsonObject=new JsonObject();
        jsonObject.addProperty("nomeSocieta",this.getNomeSocieta());
        jsonObject.addProperty("indirizzoEmail",this.getIndirizzoEmail());
        jsonObject.addProperty("password",this.getPassword());
        jsonObject.addProperty("ban",this.isBan());
        jsonObject.add("wallet", this.getWallet().toJson());
        return jsonObject;
    }


}

File: Frontend/ready2road/src/app/model/Venditore.ts
import {Wallet} from "./Wallet";
import {Inject} from "@angular/core";
import {UtenteService} from "../services/utente.service";
import {VenditoreService} from "../services/venditore.service";

export class Venditore{
    protected nomeSocieta:string;
    protected indirizzoEmail:string;
    protected password:string;
    protected ban:boolean;
    protected wallet:Wallet|undefined;

    constructor(json:any,  @Inject(VenditoreService) private service: VenditoreService){
        this.nomeSocieta=json.nomeSocieta;
        this.indirizzoEmail=json.indirizzoEmail;
        this.password=json.password;
        this.ban=json.ban;

    }
    getNomeSocieta= (): string =>{
        return this.nomeSocieta;

    }
    getIndirizzoEmail= (): string =>{
        return this.indirizzoEmail;

    }
    getPassword= (): string =>{
        return this.password;

    }

    setNomeSocieta= (nomeSocieta:string): void =>{
        this.nomeSocieta=nomeSocieta;


    }
    setIndirizzoEmail= (indirizzoEmail:string): void =>{
        this.indirizzoEmail=indirizzoEmail;


    }
    setPassword= (password:string): void =>{
        this.password=password;


    }
    getBan= (): boolean =>{
        return this.ban;
    }
    setBan= (ban:boolean): void =>{
        this.ban=ban;
    }

}
